hugo/assets/scripts/statemachine.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+
+function Preview() {
+  return (
+    <div className="placeholder-wrapper">
+     <div className='placeholder'>
+      Preview not available
+     </div>
+    </div>
+  );
+}
+
+const root = createRoot(document.getElementById("preview") as HTMLElement);
+root.render(
+  <React.StrictMode>
```

Comment:
afaik the strict mode renders the inner component twice. Within prod code you should remove it. Maybe read about it and then decide.

Reply:
According to [React's docs about strict mode](https://reactjs.org/docs/strict-mode.html), this shouldn't run in production, should be okay